fathom/src/surface/elaboration/reporting.rs

```diff
@@ -307,13 +306,18 @@ impl Message {
             Message::BooleanLiteralNotSupported { range } => Diagnostic::error()
                 .with_message("boolean literal not supported for expected type")
                 .with_labels(vec![Label::primary(file_id, *range)]),
-            Message::FailedToUnify { range, error } => {
+            Message::FailedToUnify {
+                range,
+                lhs,
+                rhs,
+                error,
+            } => {
                 use unification::{Error, RenameError, SpineError};
 
                 // TODO: Make these errors more user-friendly
                 match error {
                     Error::Mismatch => Diagnostic::error()
-                        .with_message("type mismatch")
+                        .with_message(format!("type mismatch, expected: {}, got {}", lhs, rhs))
                         .with_labels(vec![Label::primary(file_id, *range)]),
```

Comment:
Wondering what your thoughts would be on tweaking this to be something like Rust's output:

```
error[E0308]: mismatched types
   --> fathom/src/core/semantics.rs:303:27
    |
303 |                 self.eval("t_type"),
    |                           ^^^^^^^^ expected enum `core::Term`, found `str`
    |
    = note: expected reference `&core::Term<'_>`
               found reference `&'static str`
```

This would look like:

```rs
Error::Mismatch => Diagnostic::error()
    .with_message("mismatched types")
    .with_labels(vec![
        Label::primary(file_id, *range)
            .with_message(format!("type mismatch, expected `{}`, found `{}`", lhs, rhs)),
    ])
    .with_notes(vec![[
        format!("expected `{}`", lhs),
        format!("   found `{}`", rhs),
    ].join("\n")]])
```

Reply:
Ahh nice. Yes, that seems better. Will do.

Reply:
Done